packages/synapse-react-client/src/components/SynapseTable/SynapseTable.tsx

```diff
@@ -238,16 +250,24 @@ export function SynapseTable(props: SynapseTableProps) {
                       ? ExpandableTableDataCell
                       : 'td'
 
+                    const renderPlaceholder = !dataHasBeenPrefetched
+
                     return (
                       <TableDataCellElement
                         key={cell.id}
                         style={{
                           width: cell.column.getSize(),
                         }}
                       >
-                        {flexRender(
-                          cell.column.columnDef.cell,
-                          cell.getContext(),
+                        {renderPlaceholder ? (
+                          <p>
+                            <Skeleton width={'80%'} height={'20px'} />
+                          </p>
+                        ) : (
+                          flexRender(
+                            cell.column.columnDef.cell,
+                            cell.getContext(),
+                          )
```

Comment:
Delay rendering the table data cells until we have prefetched all the data to prevent the waterfall effect